src/ApiService/ApiService/Functions/QueueFileChanges.cs

```diff
@@ -47,7 +53,39 @@ public class QueueFileChanges {
         await FileAdded(_log, fileChangeEvent, lastTry);
     }
 
-    private async Async.Task FileAdded(ILogTracer log, JsonDocument fileChangeEvent, bool failTaskOnTransientError) {
+    [Function("QueueFileChangesPoison")]
+    public async Async.Task Poison(
+        [QueueTrigger(QueueFileChangesPoisonQueueName, Connection = "AzureWebJobsStorage")] string msg) {
+        var fileChangeEvent = JsonSerializer.Deserialize<JsonDocument>(msg, EntityConverter.GetJsonSerializerOptions());
+
+        _ = fileChangeEvent ?? throw new ArgumentException("Unable to parse queue trigger as JSON");
+
+        // check type first before calling Azure APIs
+        const string eventType = "eventType";
+        if (!fileChangeEvent.RootElement.TryGetProperty(eventType, out var eventTypeElement)
+            || eventTypeElement.GetString() != "Microsoft.Storage.BlobCreated") {
+            return;
+        }
+
+        const string topic = "topic";
+        if (!fileChangeEvent.RootElement.TryGetProperty(topic, out var topicElement)
+            || !_storage.CorpusAccounts().Contains(new ResourceIdentifier(topicElement.GetString()!))) {
+            return;
+        }
```

Comment:
let's put all this processing inside the try catch. A malformed message could cause an exception here and we will endup with a `QueueFileChangesPoisonPoison` queue

Reply:
Better yet, Can we  invoke `QueueFileChanges.Run` directly so we don't repeat the logic.